query-engine/core/src/interpreter/query_interpreters/read.rs

```diff
@@ -62,6 +65,8 @@ fn read_one(
                 .into())
             }
 
+            None if query.options.contains(QueryOption::ThrowOnEmpty) => record_not_found(),
```

Comment:
there is not polimorphism in here, so I would prefer to not repeat the method implementation instead of demeterizing in here. I'd leave it as it is, because the cost of not leaking the abstraction would imply repeating ourselves once per query, or if we add an abstraction to generalize that, to incur in the cost of both indirection + allocation.

It's of course debatable. BitFlags is pretty generic, actually, it's a collection in the same way Vector is. I don't see that much of a problem on leaking that abstraction, as it doesn't belong to the domain model.